packages/@ember/-internals/routing/tests/system/route_test.js

```diff
@@ -280,6 +281,19 @@ moduleFor(
       assert.deepEqual(route.serialize(model, ['post_id']), { post_id: 3 }, 'serialized correctly');
     }
 
+    ['@test returns model.id if model is a Proxy'](assert) {
+      // `key` in target should fail
+      // let model = new Proxy({ id: 3 }, {
+      //   get(target, prop) { target[prop] },
+      //   has() {
+      //     return false;
+      //   }
+      // });
```

Comment:
I put this example here in case we wanted a more general solution vs the ObjectProxy (which more closely resembles Ember-Data relationship)